src/verifiers/issuerIdentity/did/didIdentityProof.ts

```diff
@@ -55,37 +61,59 @@ const verifyV2 = async (document: v2.WrappedDocument, options: VerifierOptions):
           "MALFORMED_IDENTITY_PROOF"
         );
       const correspondingProof = document.proof.find((p) => p.verificationMethod.toLowerCase() === key.toLowerCase());
-      if (!correspondingProof) return { status: "INVALID", reason: "`id` is missing from issuer" };
+      if (!correspondingProof)
+        throw new CodedError(
+          `No proof for ${key}`,
+          OpenAttestationDidCode.MALFORMED_IDENTITY_PROOF,
+          "MALFORMED_IDENTITY_PROOF"
+        );
 
-      const { verified } = await verifySignature({
+      return verifySignature({
         merkleRoot,
         key,
         signature: correspondingProof.signature,
         did,
         resolver: options.resolver,
       });
-      return { did, status: verified ? "VALID" : "INVALID" };
     }
     throw new CodedError(
       "Issuer is not using DID identityProof type",
       OpenAttestationDidCode.INVALID_ISSUERS,
       OpenAttestationDidCode[OpenAttestationDidCode.INVALID_ISSUERS]
     );
   });
-  const signatureVerifications = await Promise.all(signatureVerificationDeferred);
-  const signedOnAll =
-    signatureVerifications.some((i) => i.status === "VALID") &&
-    signatureVerifications.every((i) => i.status === "VALID");
+  const signatureVerifications: DidVerificationStatusArray = await Promise.all(signatureVerificationDeferred);
 
-  return {
-    name,
-    type,
-    data: signatureVerifications,
-    status: signedOnAll ? "VALID" : "INVALID",
-  };
+  if (signatureVerifications.length > 0 && ValidDidVerificationStatusArray.guard(signatureVerifications)) {
+    return {
+      name,
+      type,
+      data: signatureVerifications,
+      status: "VALID",
+    };
+  }
+
+  const invalidSignature = signatureVerifications.find(InvalidDidVerificationStatus.guard);
+  if (InvalidDidVerificationStatus.guard(invalidSignature)) {
+    return {
+      name,
+      type,
+      data: signatureVerifications,
+      reason: invalidSignature.reason,
+      status: "INVALID",
+    };
+  }
+  throw new CodedError(
```

Comment:
Will reach here if the array if there is no issuer in array, not sure if we want to catch that separately.

Reply:
technically it shouldn't, if there is no issuer, it will be skipped, unless calling verify straight (but that's not how the lib is used)